### PR TITLE
Automatically increment version

### PR DESCRIPTION
**Pull request by randy-sab-roy**
Automatically bump the Mirego version when running a build (in both files).
Disclaimer: this script was barely tested and uses a clunky regex.

### CODE REVIEW

**Review by SylvainMorel — APPROVED** [review]
I don't see any obvious mistake in the regex.
tbh, I'm exceptionally bad at regex. So the above doesn't mean much.
We'll see if it works. :)

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -15,6 +15,17 @@ jobs:
...
 
+      - name: Increment Version
+        run: |
+          version=$(sed -nr "s/^.+releaseVersion.*=.*'[0-9]+\.[0-9]+\.[0-9]+\.([0-9]+)'.*$/\1/p" constants.gradle)
+          new_version=$(echo ${(l:4::0:)$((version+1))})
+          sed -ri '' "s/^(.+releaseVersion.*=.*'[0-9]+\.[0-9]+\.[0-9]+\.)[0-9]+('.*)$/\1$new_version\2/g" constants.gradle
+          sed -ri '' "s/^(.*public static final String VERSION = \"[0-9]+\.[0-9]+\.[0-9]+\.)[0-9]+(\";.*)$/\1$new_version\2/g" library/common/src/main/java/com/google/android/exoplayer2/ExoPlayerLibraryInfo.java
+          git config --global user.name "$GITHUB_ACTOR"
+          git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git commit -am "Increment Version"
```
Comment (simonlary) [inline .github/workflows/cd.yaml:26]: On pourrait mettre le nouveau numéro de version dans le message de commit ça serait nice.

**Review by simonlary — APPROVED** [review]

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (flambert) [reply .github/workflows/cd.yaml:26]: Pas juste nice, nécessaire

Comment (randy-sab-roy) [reply .github/workflows/cd.yaml:26]: On veut le full version number ou le build number Mirego? (genre 0034)

Comment (simonlary) [reply .github/workflows/cd.yaml:26]: "Increment version to 2.18.6.0032" ?

Comment (flambert) [reply .github/workflows/cd.yaml:26]: Full

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -15,6 +15,18 @@ jobs:
...
+      - name: Increment Version
+        run: |
+          version=$(sed -nr "s/^.+releaseVersion.*=.*'[0-9]+\.[0-9]+\.[0-9]+\.([0-9]+)'.*$/\1/p" constants.gradle)
+          full_version=$(sed -nr "s/^.+releaseVersion.*=.*'([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)'.*$/\1/p" constants.gradle)
+          new_version=$(echo ${(l:4::0:)$((version+1))})
+          sed -ri '' "s/^(.+releaseVersion.*=.*'[0-9]+\.[0-9]+\.[0-9]+\.)[0-9]+('.*)$/\1$new_version\2/g" constants.gradle
+          sed -ri '' "s/^(.*public static final String VERSION = \"[0-9]+\.[0-9]+\.[0-9]+\.)[0-9]+(\";.*)$/\1$new_version\2/g" library/common/src/main/java/com/google/android/exoplayer2/ExoPlayerLibraryInfo.java
+          git config --global user.name "$GITHUB_ACTOR"
+          git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git commit -am "Increment version to $full_version"
```
Comment (flambert) [inline .github/workflows/cd.yaml:27]: full_version ou new_version ? 🤔

**Pull request merged**